File: src/main/java/ac/dnd/server/common/util/LocalTimeUtils.java
package ac.dnd.server.common.util;

import static ac.dnd.server.common.util.Constant.*;
import static java.util.Objects.*;

import java.time.LocalTime;
import java.time.format.DateTimeFormatter;
import java.util.Objects;

public class LocalTimeUtils {
    private LocalTimeUtils() {
        throw new IllegalStateException("Util class.");
    }

    /**
     * {@link LocalTime} 변환
     *
     * @param source 원천문자열
     * @return "HH:mm:ss" 형식으로 변환된 {@link LocalTime}
     */
    public static LocalTime toLocalTime(String source) {
        return toLocalTime(source, FORMAT_LOCAL_TIME);
    }

    /**
     * {@link LocalTime} 변환
     *
     * @param source     원천문자열
     * @param timeFormat 변환시간형식
     * @return null or "HH:mm:ss" 형식으로 변환된 {@link LocalTime}
     */
    public static LocalTime toLocalTime(String source, String timeFormat) {
        if (isNull(source) || source.isEmpty()) {
            return null;
        }

        return LocalTime.parse(source, DateTimeFormatter.ofPattern(timeFormat));
    }

    /**
     * 문자열 변환
     *
     * @param source 원천
     * @return "HH:mm:ss" 형식으로 변환된 문자열
     */
    public static String toString(LocalTime source) {
        return toString(source, FORMAT_LOCAL_TIME);
    }

    /**
     * 문자열 변환
     *
     * @param source     원천
     * @param timeFormat 변환시간형식
     * @return '변환시간형식'으로 변환된 문자열 or null
     */
    public static String toString(LocalTime source, String timeFormat) {
        if (Objects.isNull(source)) {
            return null;
        }
        return source.format(DateTimeFormatter.ofPattern(timeFormat));
    }
}

File: src/main/kotlin/ac/dnd/server/common/util/LocalTimeUtils.kt
package ac.dnd.server.common.util

import ac.dnd.server.common.util.Constant.FORMAT_LOCAL_TIME
import java.time.LocalTime
import java.time.format.DateTimeFormatter

object LocalTimeUtils {

    /**
     * [LocalTime] 변환
     *
     * @param source 원천문자열
     * @return "HH:mm:ss" 형식으로 변환된 [LocalTime]
     */
    fun toLocalTime(source: String?): LocalTime? {
        return toLocalTime(source, FORMAT_LOCAL_TIME)
    }

    /**
     * [LocalTime] 변환
     *
     * @param source     원천문자열
     * @param timeFormat 변환시간형식
     * @return null or "HH:mm:ss" 형식으로 변환된 [LocalTime]
     */
    fun toLocalTime(source: String?, timeFormat: String): LocalTime? {
        if (source.isNullOrEmpty()) {
            return null
        }
        return LocalTime.parse(source, DateTimeFormatter.ofPattern(timeFormat))
    }

    /**
     * 문자열 변환
     *
     * @param source 원천
     * @return "HH:mm:ss" 형식으로 변환된 문자열
     */
    fun toString(source: LocalTime?): String? {
        return toString(source, FORMAT_LOCAL_TIME)
    }

    /**
     * 문자열 변환
     *
     * @param source     원천
     * @param timeFormat 변환시간형식
     * @return '변환시간형식'으로 변환된 문자열 or null
     */
    fun toString(source: LocalTime?, timeFormat: String): String? {
        if (source == null) {
            return null
        }
        return source.format(DateTimeFormatter.ofPattern(timeFormat))
    }
}